Doc/library/sqlite3.rst

```diff
@@ -1591,6 +1574,22 @@ Cursor objects
       including :abbr:`CTE (Common Table Expression)` queries.
       It is only updated by the :meth:`execute` and :meth:`executemany` methods.
 
+   .. attribute:: row_factory
+
+      Control how a row fetched from this :class:`!Cursor` is represented.
+      Set to :class:`sqlite3.Row`;
+      or a custom :term:`callable` that accepts two arguments,
+      a :class:`Cursor` object and a :class:`tuple` of the row results,
+      and returns a custom object representing an SQLite row.
+      If ``None``, a fetched row is represented as a :class:`tuple`.
```

Comment:
If we drop the `Row` part (it is a little bit cluttering, now that I re-read it), we can reorder things like this:

```suggestion
      If ``None``, a fetched row is represented as a :class:`tuple`.
      Set to a :term:`callable` that accepts two arguments,
      a :class:`Cursor` object and the :class:`!tuple` of row values,
      and returns a custom object representing an SQLite row.
```

Reply:
I definitely like putting the default ``None`` case first.

IMO its still worth at least briefly mentioning it can also be set to the included `Row` class and linking it, since that seems likely to be a common use case and makes sure readers are aware they don't necessarily have to built their own custom class for this and can just drop in an existing class, if that suits their needs—otherwise, I imagine many who might otherwise find that useful might not read further to find that out, and a brief but explicit cross-reference can go a long way as a jumping off point.

The re-ordering also exposes a bit of ambiguity as to whether "Set to a XXX" is intended as an imperative, which is somewhat unusual as it lacks a conditional followup ("...to do YYY") which is just telling users to do something without a reason, or in the past tense, which isn't in fact the case.

Therefore, I suggest:

```suggestion
      If ``None``, a fetched row is represented as a :class:`tuple`.
      Can be set to the included :class:`sqlite3.Row`;
      or a custom :term:`callable` that accepts two arguments,
      a :class:`Cursor` object and the :class:`!tuple` of row values,
      and returns a custom object representing an SQLite row.
```

Reply:
Yep, I agree on all points.

Reply:
We could perhaps drop "fetched" here as well: _"If None, a row is represented as a ..."_

Reply:
Yup, updated the suggestion (FWIW)